sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -255,3 +262,121 @@ case class Pow(left: Expression, right: Expression)
       """
   }
 }
+
+case class Round(left: Expression, right: Expression)
+  extends Expression with trees.BinaryNode[Expression] with Serializable {
+
+  def this(left: Expression) = {
+    this(left, Literal.create(0, IntegerType))
+  }
+
+  override def nullable: Boolean = left.nullable || right.nullable
+
+  override lazy val resolved =
+    childrenResolved && checkInputDataTypes().isSuccess && !DecimalType.isFixed(dataType)
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if ((left.dataType.isInstanceOf[NumericType] || left.dataType.isInstanceOf[NullType])
```

Comment:
scale of Literal(l, LongType) when ` l > Int.MaxValue || l < Int.MinValue` may considered invalid input in Hive.